terraform/main.tf

```diff
@@ -0,0 +1,264 @@
+  internal                = true
+  db_password             = var.timescaledb_password
+}
+
+/* PATCHES CONFIG */
+
+module "patches" {
+  source            = "./modules/patches"
+  name              = var.name
+  container_tag     = var.patches_container_tag
+  port              = 8081
+  cluster_id        = module.ecs_cluster.cluster_id
+  security_groups   = [aws_security_group.load_balancer.id]
+  subnets           = module.ecs_base.vpc_public_subnets
+  internal          = false
+  db_host           = module.timescaledb.db_host
+  db_port           = module.timescaledb.db_port
+  db_username       = module.timescaledb.db_username
+  db_password       = var.timescaledb_password
+  kafka_server      = split(",", aws_msk_cluster.main.bootstrap_brokers)[0]
+  kafka_topic       = "updates"
+  heimdall_endpoint = module.heimdall.internal_lb_dns_name
+  ether_endpoint    = module.ether.elb_dns_name
+}
+
+/* HEIMDALL CONFIG */
+
+module "heimdall" {
+  source           = "github.com/schramm-famm/heimdall//terraform/modules/heimdall"
+  name             = var.name
+  container_tag    = var.heimdall_container_tag
+  cluster_id       = module.ecs_cluster.cluster_id
+  vpc_id           = module.ecs_base.vpc_id
+  subnets          = module.ecs_base.vpc_public_subnets
+  private_key_cert = var.private_key_cert
+  cert             = var.cert
+  endpoints = {
+    "karen"   = module.karen.elb_dns_name
+    "ether"   = module.ether.elb_dns_name
+    "patches" = module.patches.elb_dns_name
+  }
+}
+
+/* RDS CONFIG */
+
+module "rds_instance" {
+  source          = "github.com/schramm-famm/bespin//modules/rds_instance"
+  name            = var.name
+  engine          = "mariadb"
+  engine_version  = "10.2.21"
+  port            = 3306
+  master_username = var.rds_username
+  master_password = var.rds_password
+  vpc_id          = module.ecs_base.vpc_id
+  subnet_ids      = module.ecs_base.vpc_private_subnets
+}
+
+/* ETHER CONFIG */
+
+module "ether" {
+  source          = "github.com/schramm-famm/ether//terraform/modules/ether"
+  name            = var.name
+  container_tag   = var.ether_container_tag
+  port            = 8082
+  cluster_id      = module.ecs_cluster.cluster_id
+  security_groups = [aws_security_group.load_balancer.id]
+  subnets         = module.ecs_base.vpc_private_subnets
+  internal        = true
+  db_location     = module.rds_instance.db_endpoint
+  db_username     = var.rds_username
+  db_password     = var.rds_password
+  kafka_server    = split(",", aws_msk_cluster.main.bootstrap_brokers)[0]
+  kafka_topic     = "updates"
+  karen_endpoint  = module.karen.elb_dns_name
+  efs_id          = aws_efs_file_system.ether.id
+}
+
+/* EFS CONFIG */
+
+resource "aws_efs_file_system" "ether" {}
+
+resource "aws_security_group" "efs" {
+  name        = "${var.name}_efs"
+  description = "Allow NFS traffic into EFS mount targets"
+  vpc_id      = module.ecs_base.vpc_id
+
+  ingress {
+    from_port       = 2049
```

Comment:
What do you mean you've never seen Blade Runner?

Reply:
❄️🐵